Neos.Flow/Classes/ResourceManagement/ResourceManager.php

```diff
@@ -134,6 +136,20 @@ protected function initialize()
         $this->initialized = true;
     }
 
+    protected function persistResource(PersistentResource $resource, CollectionInterface $collection, string $sourceHint = ''): void
+    {
+        $this->resourceRepository->add($resource);
+        GenericObjectValidator::onValidated($resource, function (Result $result) use ($resource) {
```

Comment:
register a validation callback that will remove the object from persistence if it had errors.